tachyon/math/finite_fields/binary_tower_operations.h

```diff
@@ -307,6 +307,69 @@ struct BinaryTowerOperations<F, std::enable_if_t<F::kBits == 8>> {
   }
 };
 
+template <typename F>
+struct BinaryTowerOperations<F, std::enable_if_t<(F::kBits >= 16)>> {
+  using SubPrimeField = typename F::SubPrimeField;
+
+  constexpr static F Mul(F lhs, F rhs) {
+    // clang-format off
+    //   (lhs₀ + lhs₁X) * (rhs₀ + rhs₁X)
+    // = lhs₀ * rhs₀ + (lhs₀ * rhs₁ + lhs₁ * rhs₀)X + (lhs₁ * rhs₁)X²
+    // = lhs₀ * rhs₀ + (lhs₀ * rhs₁ + lhs₁ * rhs₀)X + (lhs₁ * rhs₁)(αX + 1)
+    // = lhs₀ * rhs₀ + lhs₁ * rhs₁ + (lhs₀ * rhs₁ + lhs₁ * rhs₀ + lhs₁ * rhs₁ * α)X
+    // clang-format on
+    auto [lhs0, lhs1] = lhs.Decompose();
+    auto [rhs0, rhs1] = rhs.Decompose();
+    SubPrimeField z0 = lhs0 * rhs0;
+    SubPrimeField z2 = lhs1 * rhs1;
+    SubPrimeField z0z2 = z0 + z2;
+    SubPrimeField z1 = (lhs0 + lhs1) * (rhs0 + rhs1) - z0z2;
+    SubPrimeField z2a = BinaryTowerOperations<SubPrimeField>::MulByAlpha(z2);
+    return F::Compose(z0z2, z1 + z2a);
+  }
+
+  constexpr static F MulByAlpha(F x) {
+    //   (x₀ + x₁X) * X
+    // = x₀X + x₁X²
+    // = x₀X + x₁(αX + 1)
+    // = x₁ + (x₀ + x₁ * α)X
+    auto [x0, x1] = x.Decompose();
+    SubPrimeField x1a = BinaryTowerOperations<SubPrimeField>::MulByAlpha(x1);
+    return F::Compose(x1, x0 + x1a);
+  }
+
+  constexpr static F Square(F x) {
+    //   (x₀ + x₁X)²
+    // = x₀² + (2 * x₀ * x₁)X + x₁²X²
+    // = x₀² + x₁²X²
+    // = x₀² + x₁²(αX + 1)
+    // = x₀² + x₁² + (x₁² * α)X
+    auto [x0, x1] = x.Decompose();
+    SubPrimeField z0 = x0.Square();
+    SubPrimeField z2 = x1.Square();
+    SubPrimeField z2a = BinaryTowerOperations<SubPrimeField>::MulByAlpha(z2);
+    return F::Compose(z0 + z2, z2a);
+  }
+
+  constexpr static F Inverse(F x) {
+    //   (x₀ + x₁X)(x₀ + x₁ * α + x₁X)
+    // = x₀² + x₀ * x₁ * α + (x₀ * x₁)X + (x₀ * x₁)X + (x₁² * α)X + x₁²X²
+    // = x₀² + x₀ * x₁ * α + (x₁² * α)X + x₁²X²
+    // = x₀² + x₀ * x₁ * α + x₁²(αX + X²)
+    // = x₀² + x₀ * x₁ * α + x₁²
+    // = x₀(x₀ + x₁ * α) + x₁²
+    // = δ
+    auto [x0, x1] = x.Decompose();
+    SubPrimeField x0x1a =
+        x0 + BinaryTowerOperations<SubPrimeField>::MulByAlpha(x1);
+    SubPrimeField delta = x0 * x0x1a + x1.Square();
+    SubPrimeField delta_inv = delta.Inverse();
+    SubPrimeField inv0 = delta_inv * x0x1a;
+    SubPrimeField inv1 = delta_inv * x1;
+    return F::Compose(inv0, inv1);
```

Comment:
Suppose that this inverse `(x₀ + x₁ * α + x₁X) * δ⁻¹` is true, then `(x₀ + x₁X) * (x₀ + x₁ * α + x₁X) * δ⁻¹ = 1`
In other words, `(x₀ + x₁X) * (x₀ + x₁ * α + x₁X) = δ`. This is how the comment wants to explain.

Reply:
I think it would still be nice to make the first line denote the actual inverse and elaborate on later lines instead to keep up with the same structure as the previous functions as well.